### PR TITLE
fix(formulas): Support aggregated session duration

### LINKED ISSUE #15090: Formula with query including average session duration fails
## Bug description

There seems to be an issue with Formulas when one of the values uses Session Duration as an averaged property value

## How to reproduce

1. Go to this insight (Posthog employees) https://app.posthog.com/insights/0OZy62ds
2. Try to add the formula `A/B` (or actually even just `A`)
3. Crash!

Related Sentry issue (I believe) https://posthog.sentry.io/issues/3840456243/

## Environment

- [x] PostHog Cloud US, project ID: [please provide from https://app.posthog.com/project/settings#project-id]
- [ ] PostHog Cloud EU, project ID: [please provide from https://eu.posthog.com/project/settings#project-id]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see ["Sunsetting Kubernetes support"](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Resolves https://github.com/PostHog/posthog/issues/15090#issuecomment-1649693499. More details in code.

## How did you test this code?

Added a query test.

### CODE REVIEW

**Review by thmsobrmlr — APPROVED** [review]

File: posthog/queries/trends/trends_event_query_base.py [hunk 1]
```diff
@@ -52,6 +52,14 @@ def get_query_base(self) -> Tuple[str, Dict[str, Any]]:
         sample_clause = "SAMPLE %(sampling_factor)s" if self._filter.sampling_factor else ""
         self.params.update({"sampling_factor": self._filter.sampling_factor})
 
+        whitespace_padding = " " * getattr(self._entity, "index", 0)
+        # You may be wondering what the hell is this for. Well, in MOST cases, this padding has absolutely zero effect.
+        # However, unfortunately, it's critical for the success of formula queries that involve statistical math over
+        # session duration. Why exactly? No idea (https://posthog.slack.com/archives/C0113360FFV/p1690377791876449),
+        # but this is the only thing that fixes the case. It's important for ClickHouse that the whitespace is different
+        # for each entity's (variable's) SQL (hence using the index here) - a bewildering parsing bug.
+        # Remove this hack once test_regression_formula_with_session_duration_aggregation passes without it.
```
Comment (thmsobrmlr) [inline posthog/queries/trends/trends_event_query_base.py:61]:
```suggestion
        # Remove this hack once test_regression_formula_with_session_duration_aggregation passes without it.
        # See also https://github.com/ClickHouse/ClickHouse/issues/52629.
```

**Comment by Twixes** [comment]
Well, from https://github.com/ClickHouse/ClickHouse/issues/52629 it turns out that the actual issue is dollar signs not _really_ being supported in identifiers – they work _a lot_ of the time, but heredoc parsing takes precedence, which breaks in [very particular conditions](https://github.com/ClickHouse/ClickHouse/issues/52629#issuecomment-1653376969).
Reworked this PR to a sane fix, which turns out to be simply wrapping all the `$session_id` identifiers in quotes.

**Review by thmsobrmlr — APPROVED** [review]
Thumbs up for the sane fix as well. Curious if this will fix any other query issues.

Makes me think we should quote all identifiers that could contain `$` signs (internally or user generated).

**Comment by Twixes** [comment]
I think now we do. These are the only places where we used the dollar sign without quotes internally, and HogQL's `escape_clickhouse_identifier` already treats them as special.

**Pull request merged**